src/Facets/EcoFacet.sol

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+
+pragma solidity ^0.8.17;
+
+import { ILiFi } from "../Interfaces/ILiFi.sol";
+import { IEcoPortal } from "../Interfaces/IEcoPortal.sol";
+import { LibAsset } from "../Libraries/LibAsset.sol";
+import { LibSwap } from "../Libraries/LibSwap.sol";
+import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+import { SwapperV2 } from "../Helpers/SwapperV2.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+import { LiFiData } from "../Helpers/LiFiData.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { InvalidConfig, InvalidReceiver, InformationMismatch } from "../Errors/GenericErrors.sol";
+
+/// @title EcoFacet
+/// @author LI.FI (https://li.fi)
+/// @notice Provides functionality for bridging through Eco Protocol
+/// @custom:version 1.0.0
+contract EcoFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable, LiFiData {
+    /// Storage ///
+
+    // solhint-disable-next-line immutable-vars-naming
+    IEcoPortal public immutable portal;
+
+    /// Types ///
+
+    /// @dev Eco specific parameters
+    /// @param receiverAddress Address that will receive tokens on destination chain
+    /// @param nonEVMReceiver Destination address for non-EVM chains (bytes format)
+    /// @param prover Address of the prover contract for validation
+    /// @param rewardDeadline Timestamp for reward claim eligibility
+    /// @param solverReward Native token amount to reward the solver
```

Comment:
I think this is a native or ERC20 token amount, depending on sendingAssetId

Reply:
fixed: bd753870